### DISCUSSION TITLE
Some suggested improvements

### DISCUSSION THREAD

**Discussion by Agitatio** [BudgetAirpods/InsanityRemastered#23] [Ideas]
- Configurable sanity gain/loss multiplier
- Toggle for scaling sanity gain/loss by the number of players on the server. More players = faster sanity loss. Needs to be clamped in some way or have a lenient (logarithmic?) formula to avoid making big lobbies a nightmare. Should make both solo and big lobbies a decent experience. Check for players should be done every time ship gets into orbit.

**Comment by BudgetAirpods** [comment]
Thank you for the suggestion! I will add that most likely in the next update.

**Comment by Agitatio** [comment]
Going to add here to avoid opening new issue:
Seems that having flashlight on doesn't affect sanity loss. Also, it seems that after losing all of it the debuff you get is permanent. Even after gaining back all your sanity. Tested with InsanityRemastered and InsanityMeter only.

**Comment by BudgetAirpods** [comment]
Flashlights not reducing sanity has been fixed, so has the permanent debuff/slowness. Thank you for bringing both of these to my attention. (Especially the flashlight one. I thought it worked but it turns out I'm just very dumb) 


As of the latest update, I have made most sanity gain/loss multipliers configurable. I have not added in scaling with bigger lobbies yet. I have however, added a setting in the config that allows you to configure the scaling in solo play. I will make sure lobby scaling is added in the next update.

**Comment by Agitatio** [comment]
Thank you! Here's another potential enhancements I got on my mind:
- Fake scrap should only spawn when you don't have other players nearby and spawned ones should disappear when a non-hallucinating player approaches it. Both ranges ideally should be configurable.
- Pill Bottle could use a hint indicating that pressing LMB will restore your sanity.

Also, I think Solo insanity speed scaling default value is wrong. It makes you lose sanity really quick. I assume it's a multiplier, so shouldn't it be less than 1?

**Comment by Noranecore** [comment]
I would also add that the default value for sanity gain while near others is currently -0.1
This is probably wrong and will actually result in losing sanity faster while being near others, the value should be 0.1 instead

**Comment by Agitatio** [comment]
> I would also add that the default value for sanity gain while near others is currently -0.1 This is probably wrong and will actually result in losing sanity faster while being near others, the value should be 0.1 instead if you are looking

Nah. That one is correct. Positive values are how much insanity you gain, and negative ones are how much insanity you lose. I tested with two clients and having another person properly slows down insanity gain. I guess naming of the options is the source of confusion. Could just rename them all to reference insanity instead of sanity.

**Comment by BudgetAirpods** [comment]
> Thank you! Here's another potential enhancements I got on my mind:
> 
> * Fake scrap should only spawn when you don't have other players nearby and spawned ones should disappear when a non-hallucinating player approaches it. Both ranges ideally should be configurable.
> * Pill Bottle could use a hint indicating that pressing LMB will restore your sanity.
> 
> Also, I think Solo insanity speed scaling default value is wrong. It makes you lose sanity really quick. I assume it's a multiplier, so shouldn't it be less than 1?

1. I'll think about adding that. Definitely thinking about adding configurations for the spawning of them though. I'm now realizing setting customizability is going to be important for this mod, so anything major will have their values configurable in the future.

2. I actually somehow forgot about doing this. Thank you for reminding me to do something about this.

3. It's kind of a bug. I set it to that because I felt the solo insanity gain speed was too slow. Thankfully the setting is customizable, but I might end up lowering it if it's really bothering to other people. Thank you for letting me know about that one.

**Comment by BudgetAirpods** [comment]
> > I would also add that the default value for sanity gain while near others is currently -0.1 This is probably wrong and will actually result in losing sanity faster while being near others, the value should be 0.1 instead if you are looking
> 
> Nah. That one is correct. Positive values are how much insanity you gain, and negative ones are how much insanity you lose. I tested with two clients and having another person properly slows down insanity gain. I guess naming of the options is the source of confusion. Could just rename them all to reference insanity instead of sanity.

I will correct the naming of those settings. I'm not really the greatest at naming things. Sorry for the confusion, Sayounara.
